library/platform_util.c

```diff
@@ -79,6 +83,87 @@ void mbedtls_platform_zeroize( void *buf, size_t len )
 }
 #endif /* MBEDTLS_PLATFORM_ZEROIZE_ALT */
 
+void *mbedtls_platform_memset( void *ptr, int value, size_t num )
```

Comment:
Architecture - to partially answer my own question above, because we're overwriting with random data, rather than name the function `mbedtls_platform_memset()`, maybe it should be named `mbedtls_platform_erase_secret()`, and we should abstract away `memset` to allow the target to provide its own `memset`.